src/tablefilter.js

```diff
@@ -2357,24 +2357,79 @@ export class TableFilter {
      *
      * TODO: provide an API returning data in JSON format
      */
-    getFilteredDataCol(colIndex, includeHeaders = false) {
+    // getFilteredDataCol(colIndex, includeHeaders = false) {
+    //     if (isUndef(colIndex)) {
+    //         return [];
+    //     }
+    //     let data = this.getFilteredData(),
+    //         colData = [];
+    //     if (includeHeaders) {
+    //         colData.push(this.getHeadersText()[colIndex]);
+    //     }
+    //     for (let i = 0, len = data.length; i < len; i++) {
+    //         let r = data[i],
+    //             //cols values of current row
+    //             d = r[1],
+    //             //data of searched column
+    //             c = d[colIndex];
+    //         colData.push(c);
+    //     }
+    //     return colData;
+    // }
+
+    /**
+     * Return the filtered data for a given column index
+     * @param  {Number} colIndex Colmun's index
+     * @param  {Boolean} includeHeaders  Optional: include headers row
+     * @param  {Boolean} num     Optional: return unformatted number
+     * @param  {Array} exclude   Optional: list of row indexes to be excluded
+     * @param  {Boolean} visible Optional: return only visible data
+     *                           (relevant for paging)
+     * @return {Array}           Flat list of values ['val0','val1','val2'...]
+     *
+     * TODO: provide an API returning data in JSON format
+     */
+    getFilteredDataCol(
+        colIndex,
+        includeHeaders = false,
+        num = false,
+        exclude = [],
+        visible = true
+    ) {
         if (isUndef(colIndex)) {
             return [];
         }
-        let data = this.getFilteredData(),
-            colData = [];
-        if (includeHeaders) {
-            colData.push(this.getHeadersText()[colIndex]);
+
+        let rows = this.tbl.rows;
+
+        // ensure valid rows index do not contain excluded rows and row is
+        // displayed
+        let validRows = this.getValidRows(true).filter((rowIdx) => {
+            return exclude.indexOf(rowIdx) === -1 &&
+                (visible ?
+                    this.getRowDisplay(rows[rowIdx]) !== 'none' :
+                    true);
+        });
+
+        let decimal = this.decimalSeparator;
+        if (this.hasType(colIndex, [FORMATTED_NUMBER])) {
+            let colType = this.colTypes[colIndex];
+            if (colType.hasOwnProperty('decimal')) {
+                decimal = colType.decimal;
+            }
         }
```

Comment:
This painful block will hopefully disappear when we'll make the difference between `getCellData` and `getCellValue` 😄